src/linux/classes/com/alibaba/wisp/engine/WispCarrier.java

```diff
@@ -138,8 +138,8 @@ final WispTask runTaskInternal(Runnable target, String name, Thread thread, Clas
         WispTask wispTask;
         try {
             counter.incrementCreateTaskCount();
-            if ((wispTask = getTaskFromCache()) == null) {
-                wispTask = new WispTask(this, null, true, false);
+            if ((stackSize != WispConfiguration.STACK_SIZE) || (wispTask = getTaskFromCache()) == null) {
```

Comment:
I suggest putting this if statement in `getTaskFromCache()` to keep the style consistent with `returnTaskToCache()`